src/System.Private.CoreLib/shared/System/SpanHelpers.Byte.cs

```diff
@@ -959,25 +960,34 @@ public static unsafe int LastIndexOfAny(ref byte searchSpace, byte value0, byte
         public static unsafe bool SequenceEqual(ref byte first, ref byte second, nuint length)
         {
             if (Unsafe.AreSame(ref first, ref second))
+            {
                 goto Equal;
+            }
 
+            bool result = false;
             IntPtr i = (IntPtr)0; // Use IntPtr for arithmetic to avoid unnecessary 64->32->64 truncations
             IntPtr n = (IntPtr)(void*)length;
 
-            if (Vector.IsHardwareAccelerated && (byte*)n >= (byte*)Vector<byte>.Count)
+            // Conditinally jump forward for Vector versions to give the non-Vector variant used for shorter
+            // lengths an unprimed branch prediction boost
+            if (Avx2.IsSupported && 
+                (byte*)n >= (byte*)Vector256<byte>.Count)
             {
-                n -= Vector<byte>.Count;
-                while ((byte*)n > (byte*)i)
-                {
-                    if (Unsafe.ReadUnaligned<Vector<byte>>(ref Unsafe.AddByteOffset(ref first, i)) !=
-                        Unsafe.ReadUnaligned<Vector<byte>>(ref Unsafe.AddByteOffset(ref second, i)))
-                    {
-                        goto NotEqual;
-                    }
-                    i += Vector<byte>.Count;
-                }
-                return Unsafe.ReadUnaligned<Vector<byte>>(ref Unsafe.AddByteOffset(ref first, n)) ==
-                       Unsafe.ReadUnaligned<Vector<byte>>(ref Unsafe.AddByteOffset(ref second, n));
+                goto Avx2;
+            }
+
+            if (Sse2.IsSupported && 
+                (byte*)n >= (byte*)Vector128<byte>.Count)
+            {
+                goto Sse2;
+            }
+
+            if (!Avx2.IsSupported && !Sse2.IsSupported &&
```

Comment:
It would be nice if we had a tracking bug for all the places that this code is shared with, so we can completely remove the `Vector<T>` code-path....

Reply:
Arm

Reply:
Ah right, and we don't have the intrinsic support for ARM ready yet.

Reply:
Keep in mind Mono as well. They are sharing all this code.